### PR TITLE
It's impossible to use single-quote in an assertion message

### LINKED ISSUE #867: Assert which message string literal uses double-quotes with single-quotes inside fails
**Versions:**
`freezed: 2.3.2`
`freezed_annotation: 2.2.0`
`build_runner: 2.3.3`

**Describe the bug**
When using `Assert` annotation and the `message` argument is a string literal that used double-quotes with single-quotes inside, the building fails.

**To Reproduce**

Try building the following freezed class:
```dart
@freezed
class WithDoubleQuotes with _$WithDoubleQuotes {
  @Assert('false', "This assertion isn't really useful!")
  const factory WithDoubleQuotes() = _WithDoubleQuotes;
}
```

**Expected behavior**
The generated constructor should be:

```dart
const _$_WithDoubleQuotes(): assert(false, "This assertion isn't really useful!");
```
or

```dart
const _$_WithDoubleQuotes(): assert(false, 'This assertion isn\'t really useful!');
```

However, the generator tries to generate the following:
```dart
const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
```
The tentative has parsing errors, which breaks the entire building process.

The full output of trying to generate the given class is:
<details>

```
pintome@BTGDEV548NMD6M ~/P/freezed_double_quote> flutter pub get && flutter pub run build_runner build -d
Running "flutter pub get" in freezed_double_quote...             2,402ms
[INFO] Generating build script...
[INFO] Generating build script completed, took 461ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 50ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 497ms

[INFO] Running build...
[INFO] 1.7s elapsed, 0/1 actions completed.
[SEVERE] freezed on bin/freezed_double_quote.dart:
An error `FormatterException` occurred while formatting the generated source for
  `asset:freezed_double_quote/bin/freezed_double_quote.dart`
which was output to
  `bin/freezed_double_quote.freezed.dart`.
This may indicate an issue in the generator, the input source code, or in the
source formatter.
Could not format because the source could not be parsed:

line 76, column 84 of .: Unterminated string literal.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                                    ^
   ╵
line 105, column 1 of .: Expected to find ')'.
    ╷
105 │ }
    │ ^
    ╵
line 76, column 66 of .: A function body must be provided.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                  ^
   ╵
line 76, column 66 of .: Const constructors can't have a body.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                  ^
   ╵
line 76, column 68 of .: Expected to find ';'.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                    ^^^^^^
   ╵
line 76, column 75 of .: Variables must be declared using the keywords 'const', 'final', 'var' or a type name.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                           ^^^^^^
   ╵
line 76, column 75 of .: Expected to find ';'.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                           ^^^^^^
   ╵
line 76, column 81 of .: Operator declarations must be preceded by the keyword 'operator'.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                                 ^
   ╵
line 76, column 81 of .: The string '!' isn't a user-definable operator.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                                 ^
   ╵
line 76, column 81 of .: Methods must have an explicit list of parameters.
   ╷
76 │   const _$_WithDoubleQuotes(): assert(false, 'This assertion isn't really useful!');
   │                                                                                 ^
   ╵
(2 more errors...)
[INFO] Running build completed, took 2.0s

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 24ms

[SEVERE] Failed after 2.1s
pub finished with exit code 1
```
</details>

### PR DESCRIPTION
**Pull request by hajajmaor**
Fixes #867 
bump path version by 1

### CODE REVIEW

File: packages/freezed/example/lib/main.dart [hunk 1]
```diff
@@ -113,4 +114,6 @@ void main() {
   // COMPILE ERROR
   // print(example.age);
   // print(example.population);
+  final temp = WithDoubleQuotes();
```
Comment (rrousselGit) [inline packages/freezed/example/lib/main.dart:117]: As an example, this isn't quite useful.
We should have tests instead

File: packages/freezed/pubspec.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: freezed
 description: >
   Code generation for immutable classes that has a simple syntax/API without
   compromising on the features.
-version: 2.4.2
```
Comment (rrousselGit) [inline packages/freezed/pubspec.yaml:5]: Don't bump version numbers :)

**Comment by hajajmaor** [comment]
@rrousselGit Waiting for your approval or comment

File: packages/freezed/example/test/diagnosticable_test.dart [hunk 3]
```diff
@@ -65,4 +66,9 @@ void main() {
 
     expect(timeslot, isNot(isA<DiagnosticableTree>()));
   });
+
+  test('with double quotes', () {
+    final temp = const WithDoubleQuotes();
```
Comment (rrousselGit) [inline packages/freezed/example/test/diagnosticable_test.dart:71]: Shouldn't this throw?

File: packages/freezed/example/lib/main.dart [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import 'package:example/with_double_quotes.dart';
```
Comment (rrousselGit) [inline packages/freezed/example/lib/main.dart:1]:
```suggestion
```

File: packages/freezed/example/lib/main.dart [hunk 5]
```diff
@@ -113,4 +114,6 @@ void main() {
   // COMPILE ERROR
   // print(example.age);
   // print(example.population);
+  final temp = const WithDoubleQuotes();
+  print(temp);
```
Comment (rrousselGit) [inline packages/freezed/example/lib/main.dart:118]:
```suggestion
```

File: packages/freezed/example/test/diagnosticable_test.dart (hunk 3 above)
Comment (rrousselGit) [reply packages/freezed/example/test/diagnosticable_test.dart:71]: We should be getting a compilation error at the moment due to the `const` & assert(false)


Could you update the test to check that an assertion error is thrown with the correct message?

File: packages/freezed/example/lib/with_double_quotes.dart [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'with_double_quotes.freezed.dart';
```
Comment (rrousselGit) [inline packages/freezed/example/lib/with_double_quotes.dart:3]: Could you move this to packages/freezed/test/integration?

**Comment by rrousselGit** [comment]
This seems stale, so closing :)

**Pull request closed without merging**